### ISSUE TITLE
GoGradle fails on the Jenkins build agent

### ISSUE THREAD

**Issue by suyashkhandwe** [gogradle/gogradle#306]
Thanks for the awesome job !!! While we build our Go app locally, it works fine with **./gradlew goClean goBuild**, but if we run this same build on Jenkins CI node, the build fails with the following error - 

> Found go 1.12.7 in /go/bin/go, use it.
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Could not determine the dependencies of task ':resolveBuildDependencies'.
>   path must be absolute!
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

**Environment being used** - Docker image : nikerise/gradlegolang

Any pointers on what could be wrong?

**gradle-build-with-stacktrace**
[gradle-build-with-stacktrace.txt](https://github.com/gogradle/gogradle/files/3404224/gradle-build-with-stacktrace.txt)

**Comment by blindpirate** [comment]
Looking at the code, it seems like gogradle can't find an absolute gradle_user_home path. Did you overwrite gradle user home via `GRADLE_USER_HOME` or `gradle.user.home` with a relative path?

**Comment by suyashkhandwe** [comment]
Could be something with the Docker image. I will check that. Generally, what's the expected value for that environment variable?

**Comment by blindpirate** [comment]
It should not be set at all, or be set to an absolute path.

**Comment by suyashkhandwe** [comment]
I'll be investigating into it and will let you know if it works.

**Comment by suyashkhandwe** [comment]
So I tried a few experiments and I and checked the environment variables too. Here is the current failure - 

> Starting a Gradle Daemon (subsequent builds will be faster)
> 
> > Configure project :
> Found go 1.12.7 in /go/bin/go, use it.
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Could not determine the dependencies of task ':resolveBuildDependencies'.
> > path=?/.gradle/go/repo must be absolute!
> 
> * Try:
> Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Exception is:
> org.gradle.api.internal.tasks.TaskDependencyResolveException: Could not determine the dependencies of task ':resolveBuildDependencies'.
> 	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:68)

The environment variables are empty/null. This is super weird !! Is there some sort of OS dependency which I need to define somewhere?

**Comment by blindpirate** [comment]
> path=?/.gradle/go/repo must be absolute!

The problem is that gogradle can't find gradle user home. I don't know what's inside your container. There's various ways to set gradle user home: https://docs.gradle.org/current/userguide/build_environment.html

https://mrhaki.blogspot.com/2010/09/gradle-goodness-changing-gradle-user.html

Maybe you need to check if your project overwrites default gradle user home settings.

**Comment by suyashkhandwe** [comment]
Found the issue, there was an environment variable being set by the Jenkins pipeline into the docker container. Once I got hold of that, it was a matter of making sure that the environment variable value is as expected by GoGradle and configure a custom step in the pipeline to update the environment variable. Works !!

On a **side note**, why does GoGradle depend on the absolute path, can't it be using the relative path to build.gradle ?

Marking the issues closed since apart from the above clarification, there is no apparent issue in the plugin.
